src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -220,19 +221,76 @@ public NetworkTime getNetworkTime() {
     }
   }
 
-  /** Returns {@code true} if the result of spawn may be cached remotely. */
-  public boolean mayBeCachedRemotely(Spawn spawn) {
-    return remoteCache != null && Spawns.mayBeCached(spawn) && Spawns.mayBeCachedRemotely(spawn);
+  private static boolean useRemoteCache(RemoteOptions options) {
+    return !isNullOrEmpty(options.remoteCache) || !isNullOrEmpty(options.remoteExecutor);
   }
 
-  /** Returns {@code true} if the result of spawn may be cached. */
-  public boolean mayBeCached(Spawn spawn) {
-    return remoteCache != null && Spawns.mayBeCached(spawn);
+  private static boolean useDiskCache(RemoteOptions options) {
+    return options.diskCache != null && !options.diskCache.isEmpty();
```

Comment:
Can we use `!isNullOrEmpty(options.diskCache)` here?

Reply:
`options.diskCache` is a `PathFragment`, before converting to `String` we have to check for `null`, which makes `isNullOrEmpty` less appealing.